scripts/encode_data.py

```diff
@@ -42,16 +49,34 @@ def embed_documents(self) -> None:
 
             if name == "data/links":
                 continue
-
-            thread = threading.Thread(target=self._process_document, args=(file,))
-
-            self.semaphore.acquire()
-
-            thread.start()
+            
+            if self.thread:
+                threading.Thread(target=self._process_document, args=(file,)).start()
+                self.semaphore.acquire()
+            else:
+                self._process_document(file)
```

Comment:
This should stay threaded

Reply:
I put a option for it. For some reason save() happens when the threads are still running

Reply:
Interesting, blocking was working properly for me when I was running it on my machine, but now I see it's an option thats fine

Reply:
I added a option to thread because sometimes the save function is called before the threads finish running